File: commons/src/main/java/org/dhis2/commons/data/tuples/Pair.java
package org.dhis2.commons.data.tuples;

import androidx.annotation.NonNull;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class Pair<A, B> {

    @NonNull
    public abstract A val0();

    @NonNull
    public abstract B val1();

    @NonNull
    public static <A, B> Pair<A, B> create(@NonNull A val0, @NonNull B val1) {
        return new AutoValue_Pair<>(val0, val1);
    }
}

File: app/src/main/java/org/dhis2/data/tuples/Pair.kt
package org.dhis2.data.tuples

import com.google.auto.value.AutoValue

@AutoValue
abstract class Pair<A, B> {

    abstract fun val0(): A

    abstract fun val1(): B

    companion object {

        fun <A, B> create(val0: A, val1: B): Pair<A, B> {
            return AutoValue_Pair(val0, val1)
        }
    }
}
